### ISSUE TITLE
Proposal: Onboarding Flow

### ISSUE THREAD

**Issue by chrishobcroft** [batuaWallet/ux#5]
This onboarding flow is entirely aimed at localising the language used in the app, to help a user understand.

Known issues are:

- this does not actively cater for any other language or dialect than the 11 listed

- this does not cater to visually impaired, or those hard of hearing.

![PXL_20201010_070900046~2](https://user-images.githubusercontent.com/2212651/95648581-0a8a4b00-0af6-11eb-913a-8d6801ca8051.jpg)

**Comment by aditiagrwl** [comment]
I am convinced with the idea that we'll be asking the user to choose their preferred language asap while they're interacting with the app. 
However, currently the cognitive load of reading "I prefer this this language" in so many languages is too much right now. Will be taking care of it during UI. 
Also trying to associate these languages with an imagery/small icon/illustration to remove the load! Will be dropping some wireframes for this soon.

**Comment by aditiagrwl** [comment]
Information Architecture for the Onboarding flow. Thinking of Introducing a very brief guided tour of the home screen that introduces the user to the interface.

![WhatsApp Image 2020-10-11 at 11 54 16 AM](https://user-images.githubusercontent.com/72272139/95671853-007d5080-0bb9-11eb-8e9a-d9d2a59a90b5.jpeg)

**Comment by aditiagrwl** [comment]
Onboarding Flow-
Seed Phrase/Argent bit is yet to be figured out.
![WhatsApp Image 2020-10-11 at 11 54 51 AM](https://user-images.githubusercontent.com/72272139/95671882-55b96200-0bb9-11eb-9f8c-89536f5c196d.jpeg)

**Comment by chrishobcroft** [comment]
Thanks @aaditiii 

Some questions / feedback:

0. For now, let's focus on getting a simple and ultra-low-friction process to getting a user a wallet where they can receive crypto. Let's keep in mind the options for how the wallet recovery might work, but add this in once we have a working prototype which we can play with ourselves, and start to show to users.

1. Should we still show the "SEND" button if the user has no funds to send? Even plan to have it greyed out if balance is zero.

2. Are "Send" and "Receive" the right words to be using? Feels like they are just replicating MetaMask / Trust wallet's terminology, and perhaps we can bring fresh ideas in a new context?

3. Why only 3 languages offered at the beginning?

4. Why are the languages written in English in the language selector? I don't expect someone who only speaks Telugu would do well on this.

5. Perhaps avoid any kind of text on the splash screen... the example shows Hindi and English languages, and I might be inclined to delete the app if I'm not a native Hindi or English speaker. Things like this make me wonder about whether we have tendency to want to serve English / Hindi speakers first, as they tend to be the wealthier elements of society (very broadly speaking, of course)... which is a common mindset in the Banking / Web 2.0 world where customers are primarily "to be profited from" (hence a drive to find rich ones)... instead of a Web 3.0 mindset, which gains no benefit from having wealthier customers, instead being more motivated to include everyone.

6. I don't really understand how the flow works for the actual app usage... but one question I have is why do we need the Receive button at all? Instead of "including it just because all other apps have it", try to "start without it, and then find a good reason to add it" (if absolutely necessary).

I would propose focussing on the following right now:

a. Propose a very limited onboarding flow, without anything about wallet recovery (for now), but containing functionality to allow a user to fulfil the following use cases:
  - As a user, I want to be able to understand the language used by the app
  - As a user, I want to understand a little bit more about what Batua is

b. Propose a _minimal_ functionality to allow a user to fulfil the following use cases:
  - As a user, I want to see my balance
  - As a user, I want to know how to receive money
  - As a user, I want to send money

**Comment by chrishobcroft** [comment]
> However, currently the cognitive load of reading "I prefer this this language" in so many languages is too much right now. 

I don't agree. When I see things like this, I personally feel very little cognitive load from language(s) which I _don't_ understand (which is the majority of them)... I simply ignore them, and choose a language that I _do_ understand.

I can agree that the way it is presented in my ugly picture definitely appears like "too much information", so I would welcome ways to simplify this as a UI - perhaps even just showing the names of the languages (without the ugly "I prefer ...........").

I also think that showing only three (most popular?) languages is exclusionary, and would be offputting for anyone whose primary language is not one of those three. We have a great opportunity here, to impress and delight users whose first language is _not_ Hindi / English... by a) showing that we've actually thought about them (instead of the usual habit of prioritising apps for English then Hindi mediums), and b) providing them a service in their own native language.

So... for me, it's our obligation to try to find a way to allow speakers of the 11 languages from the research (or maybe even more) to easily select their preferred language. If this means that the copy in the app needs to be slimmed down to make it easier to cover all languages, then I think this is what we should do.

But the moment we appear to be prioritising any one language over any other, we risk appearing _exclusive_, when a core tenet of this project is that we want to be _inclusive_.

From the project page: "बटुआ (Batua) is a project for all users of the Indian Rupee. It is aimed primarily at all users of Indian Rupee, irrespective of their financial wealth, language, gender, religion, caste, ethnicity and skin-colour."

**Comment by aditiagrwl** [comment]
Thanks for the feedback Chris!

1. Absolutely. The Send Button might be disabled when the balance is zero. But I'm afraid that when someone installs Batua, they're balance will be zero and having it disabled might have a negative feeling. Perhaps we can elliminate the button at that stage.

2. Yes I'm looking for alternative words. These are depictions right now.

3. No all 11 languages would be visible. This is just a per sketch and thats why it's visible like this. It just explains the concept of it.

4. I agree. I was just conceptualising this so didnt write the languages in the particular language alphabet. In the app ALL LANGUAGES would be written in their own language alphabet.

6. If we dont give the receive button to go the screen where they can see their qr code and ID, we'll have to give everything like the qr code, their ID, their balance, send button etc on one screen which in my view isnt ideal. Maybe we can discuss this once I prepare a mock

Working on the basic functionality for now. Thanks!

**Comment by aditiagrwl** [comment]
> > However, currently the cognitive load of reading "I prefer this this language" in so many languages is too much right now.
> 
> I don't agree. When I see things like this, I personally feel very little cognitive load from language(s) which I _don't_ understand (which is the majority of them)... I simply ignore them, and choose a language that I _do_ understand.
> 
> I can agree that the way it is presented in my ugly picture definitely appears like "too much information", so I would welcome ways to simplify this as a UI - perhaps even just showing the names of the languages (without the ugly "I prefer ...........").
> 
> I also think that showing only three (most popular?) languages is exclusionary, and would be offputting for anyone whose primary language is not one of those three. We have a great opportunity here, to impress and delight users whose first language is _not_ Hindi / English... by a) showing that we've actually thought about them (instead of the usual habit of prioritising apps for English then Hindi mediums), and b) providing them a service in their own native language.
> 
> So... for me, it's our obligation to try to find a way to allow speakers of the 11 languages from the research (or maybe even more) to easily select their preferred language. If this means that the copy in the app needs to be slimmed down to make it easier to cover all languages, then I think this is what we should do.
> 
> But the moment we appear to be prioritising any one language over any other, we risk appearing _exclusive_, when a core tenet of this project is that we want to be _inclusive_.
> 
> From the project page: "बटुआ (Batua) is a project for all users of the Indian Rupee. It is aimed primarily at all users of Indian Rupee, irrespective of their financial wealth, language, gender, religion, caste, ethnicity and skin-colour."

As mentioned earlier, this is just a paper sketch and I dint fit all languages here but all 11 languages would definitely be here from the start. I apologise that the idea wasnt clear in my paper sketch. Also, the languages would be written in the native language itself. We won't be prioritising/ promoting a certain language.

**Comment by chrishobcroft** [comment]
Super! For information, here's the names of the 11 different languages _expressed in each language_.

| Language Name (in English) | Language Name (in Language) |
|----------------------------|-----------------------------|
| English                    | English                     |
| Hindi                      | हिंदी                        |
| Gujarati                   | ગુજરાતી                      |
| Bengali                    | বাংলা                     |
| Urdu                       | اُردُو                     |
| Tamil                      | தமிழ்                    |
| Punjabi                    | ਪੰਜਾਬੀ                    |
| Malayalam                  | മലയാളം               |
| Marathi                    | मराठी                     |
| Kannada                    | ಕನ್ನಡ                     |
| Telugu                     | తెలుగు                     |

When expressed like this, it doesn't feel too much cognitive load
